ocaml/xapi/context.ml

```diff
@@ -234,17 +225,46 @@ let parent_of_origin (origin : origin) span_name =
   | _ ->
       None
 
-let start_tracing_helper parent_fn task_name =
+let make_attributes ?task_name ?task_id ?task_uuid ?session_id ?origin () =
+  let attribute_helper_fn f v = Option.fold ~none:[] ~some:f v in
+  [
+    attribute_helper_fn
+      (fun task_name -> [("xs.xapi.task.name", task_name)])
+      task_name
+  ; attribute_helper_fn
+      (fun task_id -> [("xs.xapi.task.id", Ref.really_pretty_and_small task_id)])
+      task_id
+  ; attribute_helper_fn
+      (fun task_uuid -> [("xs.xapi.task.uuid", Uuidx.to_string task_uuid)])
+      task_uuid
+  ; attribute_helper_fn
+      (fun session_id -> [("xs.xapi.session.id", Ref.string_of session_id)])
```

Comment:
You can't include session refs, as they are security (auth) tokens. You could translate them into track ids, as done for logging elsewhere in this module.